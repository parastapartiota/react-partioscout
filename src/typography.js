import React from 'react';
import styled, { css, injectGlobal } from 'styled-components';

export const headingStyles = css`
  @import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:900,900i&subset=latin-ext');
  font-family: 'Source Sans Pro', sans-serif;
  font-weight: black;
  color: ${props => props.color};
`;

export const textStyles = css`
  @import url('https://fonts.googleapis.com/css?family=PT+Sans:400,400i,700,700i&subset=latin-ext');
  font-family: 'PT Sans', sans-serif;
  font-weight: regular;
  color: ${props => props.color};
`;

export const PartioScoutTypography = ({ el = 'div', ...props }) => {
  const Container = styled[el]`
    body,
    * {
      ${textStyles};
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      ${headingStyles};
    }
  `;

  return <Container {...props} />;
};

export const Heading = ({ el = 'h1', ...props }) => {
  const Styled = styled[el](headingStyles);

  return <Styled {...props} />;
};

export const Text = ({ el = 'p', ...props }) => {
  const Styled = styled[el](textStyles);

  return <Styled {...props} />;
};

export const injectPartioScoutStyles = () => {
  injectGlobal`
  body, * {
    ${textStyles}
  }

  h1, h2, h3, h4, h5, h6 {
    ${headingStyles}
  }
  `;
};
