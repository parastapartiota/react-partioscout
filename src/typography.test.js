import React from 'react';
import renderer from 'react-test-renderer';
import * as T from './typography';

describe('PartioScoutTypography', () => {
  it('is truthy', () => {
    expect(T.PartioScoutTypography).toBeTruthy();
  });

  it('renders <div> by default', () => {
    const wrapper = renderer
      .create(<T.PartioScoutTypography>test</T.PartioScoutTypography>)
      .toJSON();
    expect(wrapper.type).toEqual('div');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });

  it('can render <main>', () => {
    const wrapper = renderer
      .create(<T.PartioScoutTypography el="main">test</T.PartioScoutTypography>)
      .toJSON();
    expect(wrapper.type).toEqual('main');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });
});

describe('Heading', () => {
  it('is truthy', () => {
    expect(T.Heading).toBeTruthy();
  });

  it('renders <h1> by default', () => {
    const wrapper = renderer.create(<T.Heading>test</T.Heading>).toJSON();
    expect(wrapper.type).toEqual('h1');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });

  it('can render <h2>', () => {
    const wrapper = renderer
      .create(<T.Heading el="h2">test</T.Heading>)
      .toJSON();
    expect(wrapper.type).toEqual('h2');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });
});

describe('Text', () => {
  it('is truthy', () => {
    expect(T.Heading).toBeTruthy();
  });

  it('renders <p> by default', () => {
    const wrapper = renderer.create(<T.Text>test</T.Text>).toJSON();
    expect(wrapper.type).toEqual('p');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });

  it('can render <div>', () => {
    const wrapper = renderer.create(<T.Text el="div">test</T.Text>).toJSON();
    expect(wrapper.type).toEqual('div');
    expect(wrapper.props.className).toBeDefined();
    expect(wrapper.children).toMatchSnapshot();
  });
});
