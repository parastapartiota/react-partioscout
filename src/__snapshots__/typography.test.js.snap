// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading can render <h2> 1`] = `
Array [
  "test",
]
`;

exports[`Heading renders <h1> by default 1`] = `
Array [
  "test",
]
`;

exports[`PartioScoutTypography can render <main> 1`] = `
Array [
  "test",
]
`;

exports[`PartioScoutTypography renders <div> by default 1`] = `
Array [
  "test",
]
`;

exports[`Text can render <div> 1`] = `
Array [
  "test",
]
`;

exports[`Text renders <p> by default 1`] = `
Array [
  "test",
]
`;
